src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -1520,7 +1520,9 @@ bool QgsPostgresProvider::hasSufficientPermsAndCapabilities()
       return false;
     }
 
-    bool inRecovery = false;
+    // Do not check the editable capabilities if the flag has been set
+    // or if the database crashed or connected to a standby server
+    bool inRecovery = ( mReadFlags & QgsDataProvider::ForceReadOnly );
```

Comment:
I think that you are abusing `inRecovery` here: either use a separate bool or rename it to indicate that the variable carries a different meaning than the status of the recovery.